openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -98,3 +107,32 @@ def send_notifications(user_ids, course_key, app_name, notification_type, contex
             ))
     # send notification to users but use bulk_create
     Notification.objects.bulk_create(notifications)
+
+
+def update_user_preference(preference: CourseNotificationPreference, user, course_id):
+    """
+    Update user preference if config version is changed.
+    """
+    ver = get_course_notification_preference_config_version()
+    if preference.config_version != ver:
+        return preference.get_updated_user_course_preferences(user, course_id)
+    return preference
+
+
+def create_notification_pref_if_not_exists(user_ids, preferences):
+    """
+    Create notification preference if not exist.
+    """
+    new_preferences = []
+
+    for user_id in user_ids:
+        if not any(preference.user_id == user_id for preference in preferences):
```

Comment:
We can get `values_list(user_id, flat=True)` here and then simply check using `in`

Reply:
preferences is list so we cannot use values_list here .